File: librarymanager/librarymanager-be/src/main/java/ru/spring5/elena/librarymanager/api/DictionaryApi.java
package ru.spring5.elena.librarymanager.api;

import java.util.List;

import javax.validation.constraints.NotNull;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import ru.spring5.elena.librarymanager.api.dto.Edition;
import ru.spring5.elena.librarymanager.api.dto.Genre;
import ru.spring5.elena.librarymanager.api.dto.Lang;

public interface DictionaryApi {

	Genre getGenre(
			@NotNull  Long genreId
	);

	List<Genre> searchGenres();

	Page<Edition> searchEditionsByGenreId(
			@NotNull  Long genreId, 
			Pageable pageable
	);

	Lang getLang(
			@NotNull Long langId
	);

	List<Lang> searchLangs();

	Page<Edition> searchEditionsByLangId(
			@NotNull Long langId, 
			Pageable pageable
	);

	Page<Edition> searchEditionsByPublLangId(
			@NotNull Long langId,
            Pageable pageable
    );

	List<Edition> updateEditionsByGenreId(
			@NotNull Long genreId, 
			@NotNull List<Long> editionIds
	);

	List<Edition> updateEditionsByLangId(
			@NotNull Long langId, 
			@NotNull List<Long> editionIds
	);

	List<Edition> updateEditionsByPubllangId(
			@NotNull Long publlangId, 
			@NotNull List<Long> editionIds
	);
}

File: librarymanager-api/src/main/java/ru/elenacherev/librarymanager/api/DictionaryApi.kt
package ru.elenacherev.librarymanager.api

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import ru.elenacherev.librarymanager.api.dto.Edition
import ru.elenacherev.librarymanager.api.dto.Genre
import ru.elenacherev.librarymanager.api.dto.Lang
import java.util.*

interface DictionaryApi {
    fun getGenre(
        genreId: UUID
    ): Genre?

    fun searchGenres(): List<Genre>

    fun searchEditionsByGenreId(
        genreId: UUID,
        pageable: Pageable
    ): Page<Edition>

    fun getLang(
        langId: UUID
    ): Lang?

    fun searchLangs(): List<Lang>

    fun searchEditionsByLangId(
        langId: UUID,
        pageable: Pageable
    ): Page<Edition>

    fun searchEditionsByPublLangId(
        publLangId: UUID,
        pageable: Pageable
    ): Page<Edition>

    fun updateEditionsByGenreId(
        genreId: UUID,
        editionIds: List<UUID>
    ): List<Edition>

    fun updateEditionsByLangId(
        langId: UUID,
        editionIds: List<UUID>
    ): List<Edition>

    fun updateEditionsByPubllangId(
        publLangId: UUID,
        editionIds: List<UUID>
    ): List<Edition>
}